pkg/gohai/utils/value_test.go

```diff
@@ -0,0 +1,46 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewValue(t *testing.T) {
+	value := NewValue(42)
+	v, err := value.Value()
+	require.NoError(t, err)
+	require.Equal(t, 42, v)
+}
+
+func TestNewErrorValue(t *testing.T) {
+	myErr := errors.New("this is an error")
+	value := NewErrorValue[int](myErr)
+	result, err := value.Value()
+	require.ErrorIs(t, err, myErr)
+	require.Equal(t, 0, result)
+
+	myOtherErr := errors.New("this is another error")
+	value = value.NewErrorValue(myOtherErr)
```

Comment:
It's not overwriting the error, it's actually creating a new `Value[T]` (which is why the test does `value = ...`).

`NewErrorValue` is defined in two different ways:
```go
func NewErrorValue[T any](err error) Value[T]
func (Value[T]) NewErrorValue(err error) Value[T]
```

The use case for the second one is that in `Initialize` I go over each field and initialize them with a default `Value[T]`.
However `reflect` currently doesn't play really well with generics, and I couldn't find how to call `NewErrorValue[T]` "with the same T as the one from this field".
The only workaround I could think of was to define this `NewErrorValue`, which is not exactly generic, because T is linked to the receiver type and not the function itself, it's just a method on the given field type so it can be called easily with `reflect`.